docs/ci-cd-integration-guide.md

```diff
@@ -4,7 +4,92 @@ This guide provides examples for running **Hydrophone** in CI/CD pipelines, incl
 
 ---
 
-## 1. GitHub Actions Example
+## 1. Introduction
+
+This guide covers GitHub Actions, GitLab CI, Jenkins Pipeline, Prow jobs, tips, and beginner-friendly examples to help you integrate Hydrophone into your CI/CD system.
+
+---
+
+## 2. Beginner-Friendly Examples
+
+These snippets let you run Hydrophone in CI/CD pipelines **without needing a full Kubernetes setup**.  
+You can copy–paste them directly.
+
+---
+
+## 3. GitHub Actions
+
+Create `.github/workflows/hydrophone.yml`:
+
+```yaml
+name: Hydrophone Quick Test
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  hydrophone:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+      - name: Run Hydrophone (dry run)
+        run: |
+          go run main.go --dry-run
+```
+
+---
+
+## 4. GitLab CI
+
+Add this to `.gitlab-ci.yml`:
+
+```yaml
+stages:
+  - test
+
+hydrophone:
+  stage: test
+  image: golang:1.21
```

Comment:
Our go.mod requires 1.24+ so all the image tags need to reflect that or `:latest` though specific versions would be preferred.